From 961020251b58b6dfb314a96f2f091e3fe659f09d Mon Sep 17 00:00:00 2001
From: Leo Franchi <lfranchi@kde.org>
Date: Thu, 14 Jun 2012 21:53:39 +0200
Subject: [PATCH] Set Content-Type as Qt 4.8 changes default to
 application/octet-stream

Previous default was application/x-www-form-urlencoded which is what we want. Without this, all POSTS fail that expect that (such as handshaking)
---
 src/ws.cpp |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/ws.cpp b/src/ws.cpp
index 196af40..54bade9 100644
--- a/src/ws.cpp
+++ b/src/ws.cpp
@@ -171,8 +171,9 @@ lastfm::ws::post( QMap<QString, QString> params, bool sk )
                + QUrl::toPercentEncoding( i.value() )
                + '&';
     }
-
-    return nam()->post( QNetworkRequest(baseUrl()), query );
+    QNetworkRequest req( baseUrl() );
+    req.setHeader( QNetworkRequest::ContentTypeHeader, "application/x-www-form-urlencoded" );
+    return nam()->post( req, query );
 }
 
 
-- 
1.7.10.4

